usermods/sht/usermod_sht.h

```diff
@@ -191,7 +191,7 @@ void ShtUsermod::publishHomeAssistantAutodiscovery() {
   json[F("stat_cla")] = F("measurement");
   snprintf_P(buf, 127, PSTR("%s-temperature"), escapedMac.c_str());
   json[F("uniq_id")] = buf;
-  json[F("unit_of_meas")] = F("°C");
+  json[F("unit_of_meas")] = F((unitOfTemp ? "°F" : "°C"));
```

Comment:
I am not sure this is the correct way of using F().
To be on the safe side use `unitOfTemp ? F("°F") : F("°C")`

And there are some other similar statements `unitOfTemp ? ... : ...` that I find suboptimal regarding code size.

Reply:
What would be different? In both cases it passes either "°C" or "°F". the ternary operator is solved before calling F()

Reply:
Please check the definition of F() wrapper. It is not a regular function.

Reply:
I agre with @blazoncek ... F() is used to make sure that strings are read directly from flash memory. It's actually a compile-time thing, so probably will not have the desired effect if the parameter of F() is not a string constant but a condition. Also 8266 has a different Implementation of F() it might work on esp32 but maybe not work on ESP8266.

Better to change as Blaz suggested.

Reply:
Yeah, think it makes sense. I was under the assumption that the ternary operator is resolved before being passed on. But I missed that F() is actually a preprocessor macro, hence the ternary operator is most likely not resolved at the time, as this would be done at runtime, if I remember / understand correctly. Thanks for pointing it out, addressed it!